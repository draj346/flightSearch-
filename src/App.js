import React, { useState } from 'react';
import './App.css';
import SearchContainer from './components/SearchForm';
import RefineSeachForm from './components/RefineFlightSearchForm';
import styles from './styles/styles';
import FlightList from './components/FlightList';


const App = () => {

  const [searchCriteria, setSearchCriteria] = useState({
    origin: '',
    destination: '',
  });
  const [price, setPrice] = useState(0);

  return (
    <div style={appStyle.mainDivStyle}>
      <header><h2>Flight Search Engine</h2></header>
      <div style={appStyle.bottomDivStyle}>
        <div style={appStyle.leftDivStyle}>
          <SearchContainer setSearchCriteria={setSearchCriteria}></SearchContainer>
          <RefineSeachForm setPrice={setPrice}></RefineSeachForm>
        </div>
        {
          searchCriteria.origin && searchCriteria.destination ?
            <div style={appStyle.rightDivStyle}>
            <header style={appStyle.headerStyle}>
              <h3 style={appStyle.headerH3Style}>{searchCriteria.origin ? searchCriteria.origin + '>' : ''} {searchCriteria.destination}</h3>
            </header>
            <div style={appStyle.searchDisplayContainer}>
              <FlightList {...searchCriteria} price={price}></FlightList>
            </div>
          </div> :
          <div style={appStyle.rightDivStyle}>
            <h1 style={styles.noDataStyle}>No Flight Searched</h1>
          </div>
        }
      </div>
    </div>
  );
}

const appStyle = {
  mainDivStyle: {
    ...styles.borderGrey,
    margin: 5,
    width: 800,
  },
  bottomDivStyle: {
    borderTop: `1px grey solid`,
    display: 'flex',
    minHeight: 500,
  },
  leftDivStyle: {
    borderRight: `1px grey solid`,
  },
  rightDivStyle: {
    flexGrow: 1,
  },
  headerStyle: {
    borderBottom: `1px grey solid`,
  },
  headerH3Style: {
    paddingLeft: 15,
  },
  filterSection: {
    overflow: scroll,
    height: 500,
  },
  searchDisplayContainer: {
    height: 500,
    overflow: 'scroll',
  }
};

export default App;
