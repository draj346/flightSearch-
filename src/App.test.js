import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/No Flight Searched/i);
  expect(linkElement).toBeInTheDocument();
});
