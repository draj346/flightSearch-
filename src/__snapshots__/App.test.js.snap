// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  style={
    Object {
      "border": "1px grey solid",
      "margin": 5,
      "width": 800,
    }
  }
>
  <header>
    <h2>
      Flight Search Engine
    </h2>
  </header>
  <div
    style={
      Object {
        "borderTop": "1px grey solid",
        "display": "flex",
        "minHeight": 500,
      }
    }
  >
    <div
      style={
        Object {
          "borderRight": "1px grey solid",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "margin": 5,
            "textAlign": "start",
          }
        }
      >
        <span
          style={
            Object {
              "alignSelf": "flex-start",
              "borderLeft": "1px grey solid",
              "borderRight": "1px grey solid",
              "borderTop": "1px grey solid",
              "fontSize": 13,
              "fontWeight": 600,
              "paddingLeft": 5,
              "paddingRight": 5,
            }
          }
        >
          One Way
        </span>
        <div
          style={
            Object {
              "alignContent": "flex-start",
              "border": "1px grey solid",
              "display": "flex",
              "flexDirection": "column",
            }
          }
        >
          <div
            style={
              Object {
                "margin": 5,
              }
            }
          >
            <input
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter Origin City"
              role="combobox"
              value=""
            />
          </div>
          <div
            style={
              Object {
                "margin": 5,
              }
            }
          >
            <input
              aria-autocomplete="list"
              aria-expanded={false}
              autoComplete="off"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Enter Destination City"
              role="combobox"
              value=""
            />
          </div>
          <input
            placeholder="Departure Date"
            style={
              Object {
                "margin": 5,
              }
            }
            type="date"
          />
          <select
            style={
              Object {
                "margin": 5,
              }
            }
          >
            <option
              value="Passengers"
            >
              Passengers
            </option>
            <option
              value={1}
            >
              1
            </option>
            <option
              value={2}
            >
              2
            </option>
            <option
              value={3}
            >
              3
            </option>
            <option
              value={4}
            >
              4
            </option>
            <option
              value={5}
            >
              5
            </option>
          </select>
          <button
            onClick={[Function]}
            style={
              Object {
                "alignSelf": "flex-start",
                "margin": 5,
              }
            }
          >
            Search
          </button>
        </div>
      </div>
      <div
        style={
          Object {
            "border": "1px grey solid",
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      >
        <span
          style={
            Object {
              "fontSize": 13,
              "fontWeight": 600,
              "paddingLeft": 5,
            }
          }
        >
          Refine Flight Search
        </span>
        <div
          aria-orientation="horizontal"
          aria-valuemax={20000}
          aria-valuemin={0}
          aria-valuenow={0}
          className="rangeslider rangeslider-horizontal"
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="rangeslider__fill"
            style={
              Object {
                "width": "0px",
              }
            }
          />
          <div
            className="rangeslider__handle"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "left": "0px",
              }
            }
            tabIndex={0}
          >
            <div
              className="rangeslider__handle-label"
            >
              
            </div>
          </div>
          <ul
            className="rangeslider__labels"
          />
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <h1
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        No Flight Searched
      </h1>
    </div>
  </div>
</div>
`;
